client/app/core/authorization.config.js

```diff
@@ -61,16 +65,18 @@ export function authInit($rootScope, $state, Session, $sessionStorage, logger, L
 
     // user is required and session not active - not going anywhere right away
     event.preventDefault();
-
     $sessionStorage.$sync();  // needed when called right on reload
     if (!$sessionStorage.token) {
       // no saved token, go directly to login
       $state.transitionTo('login');
 
       return;
     }
-
-    // trying saved token..
+    syncSession().then(rbacReloadOrLogin(toState, toParams));
+  }
+  function syncSession() {
+    var deferred = $q.defer();
```

Comment:
For new code I think we should avoid using the `$q` service and use the native ES6 `Promise` instead.

The same 3 lines in this function using `$q` could be rewritten as `.then(Promise.resolve())` in the last part of the `then` chain.

Also, after looking this over I think you can avoid adding the extra `.then` step and instead return the Promise chain directly:

```javascript
return Session.loadUser()
  .then(Language.onReload)
  .then(ServerInfo.set)
  .then(ProductInfo.set)
  .catch(badUser);
```

In either case there is no longer any `.catch` for `rbacReloadOrLogin` where there was before.